cf-custom-resources/lib/nlb-cert-validator-updater.js

```diff
@@ -400,6 +412,135 @@ async function activateOption(option, loadBalancerDNS, loadBalancerHostedZone) {
     }).promise();
 }
 
+/**
+ * Retrieve all certificates used for the service and cache the results.
+ * @returns {Array<Object>} An array of descriptions for the certificates used by the service.
+ */
+async function serviceCertificates() {
+    if (cachedServiceCertificates) {
+        return cachedServiceCertificates;
+    }
+    let { ResourceTagMappingList } = await clients.resourceGroupsTaggingAPI().getResources({
```

Comment:
Just a callout but we used to have certificates untagged for a long time until several months ago. I guess the worst scenario is they would need to remove the orphaned certificate by themselves?

Reply:
This script is NLB specific, so it will not touch any ALB certificate. All NLB shall be tagged as such so the historically orphaned certificate wouldnt' be a concern for this particular custom resource!